tensorflow/python/ops/variable_scope.py

```diff
@@ -690,6 +695,31 @@ def _get_single_variable(self,
     return v
 
 
+  # Initialize variable when no initializer provided
+  def _get_default_initializer(self, shape=None, dtype=dtypes.float32):
+    """Use this function to provide a default and corresponding initializer of dtype
+
+    Args:
+      dtype: Type of the new or existing variable.(defaults to `DT_FLOAT`).
+
+    Returns:
+      initializer and initializing_from_value. See documentation of get_variable above.
+
+    Raises:
+      TypeError: When giving unsupported dtype.
+    """
+    if dtype.is_floating:
+      initializer = init_ops.uniform_unit_scaling_initializer()
+      initializing_from_value = False
+    elif dtype.is_integer or dtype.is_unsigned or dtype.is_bool or dtype.is_complex:
```

Comment:
since the complex case is ambiguous enough for us to have a discussion about what value it should be, let's just fall back on raising an error in the complex case.